internal/dag/gatewayapi_processor.go

```diff
@@ -61,6 +63,34 @@ func (p *GatewayAPIProcessor) Run(dag *DAG, source *KubernetesCache) {
 
 	for _, listener := range p.source.gateway.Spec.Listeners {
 
+		var listenerSecret *Secret
+		var err error
+
+		// Check for TLS on the Gateway.
+		if tls := listener.TLS; tls != nil {
+			// Validate the CertificateRef is configured.
```

Comment:
you should also check for the listener protocol. The protocol != "HTTP". Check out similar validation that I planned for the operator:

https://github.com/projectcontour/contour-operator/pull/237/files#diff-feefa601b9dc749f80c19b5769691942730947a8a8226c6ff15767916e58980cR144-R171

Also, [the validation pkg](https://github.com/kubernetes-sigs/gateway-api/pull/597) has support upstream... just working through the details. I think it makes sense for Contour to have a validation pkg that includes these checks. When the upstream validation is ready, the contour validation pkg can dump all the local validation code except any implementation-specific validation. Thoughts?

Reply:
Yup sounds good! I added support to check for HTTPS protocol since that's what we're supporting in this PR and opened another issue to track the other bits: https://github.com/projectcontour/contour/issues/3540